aspnetcore/release-notes/aspnetcore-9/includes/signalr-trimming-aot.md

```diff
@@ -0,0 +1,26 @@
+### SignalR supports trimming and native AOT
+
+Continuing the [native AOT journey](https://learn.microsoft.com/aspnet/core/fundamentals/native-aot) started in .NET 8, we have enabled trimming and native ahead-of-time (AOT) compilation support for both SignalR client and server scenarios. You can now take advantage of the performance benefits of using native AOT in applications that use SignalR for real-time web communications.
+
+#### Getting started
+
+Install the latest [.NET 9 SDK](https://get.dot.net/9).
+
+Create a solution from the `webapiaot` template in a command shell using the following command:
+
+```dotnetcli
+dotnet new webapiaot -o SignalRChatAOTExample
+```
+
+Replace the contents of the `Program.cs` file with the following SignalR code:
+
+[!code-csharp[](~/release-notes/aspnetcore-9/samples/SignalRChatAOTExample/Program.cs)]
+
+The preceding example produces a native Windows executable of 10 MB and a Linux executable of 10.9 MB.
+
+#### Limitations
+
+* Only the JSON protocol is currently supported. As shown previously, apps need to use the System.Text.Json source generator. This follows the same approach as Minimal APIs.
+* On a SignalR server, Hub method parameters of type `IAsyncEnumerable<T>` and `ChannelReader<T>` where `T` is a ValueType (i.e. `struct`) aren't supported. These will cause runtime exceptions at startup during development and in the published app. For more information, see [issue #56179](https://github.com/dotnet/aspnetcore/issues/56179).
+* Client proxy generation is not supported in PublishAot. Client proxy generation will result in warnings at build-time and works with `PublishTrimmed` with no warnings.
```

Comment:
> Client proxy generation will result in warnings at build-time and works with `PublishTrimmed` with no warnings.

This feels like a run-on sentence. Can it be split into 2 sentences?

Reply:
Sure, thanks!

Reply:
"Client proxy generation" -> [Strongly-typed hubs](https://learn.microsoft.com/aspnet/core/signalr/hubs?view=aspnetcore-8.0#strongly-typed-hubs)

Reply:
@BrennanConroy, by this do you mean something like the following should be added to clarify the relationship to Strongly-typed hubs?

"Client proxy generation in ASP.NET Core SignalR enables the creation of strongly-typed hubs, ensuring type-safe communication between clients and server-side hubs.

Reply:
No, replace "client proxy generation" with "strongly-typed hubs"